File: mutability.ex
defmodule MutationDemo do
  def mutate_list(my_list)  do
    [ "tunecore" | my_list ]
  end

  def mutate_map(my_map) do
    Map.merge my_map, %{true_name: "Rumplestiltskin", age: 130}
  end
end

File: mutability.rb
module MutationDemo
  def self.mutate_list(my_list)
    my_list.unshift("tunecore")
  end

  def self.mutate_hash(my_hash)
    my_hash.merge!({true_name: "Rumplestiltskin", age: 130})
  end
end
